libraries/AP_Param/AP_Param.cpp

```diff
@@ -2429,14 +2424,30 @@ void AP_Param::load_embedded_param_defaults(bool last_pass)
 /* 
    find a default value given a pointer to a default value in flash
  */
-float AP_Param::get_default_value(const AP_Param *vp, const float *def_value_ptr)
+float AP_Param::get_default_value(const AP_Param *vp, const struct GroupInfo &info)
 {
     for (uint16_t i=0; i<num_param_overrides; i++) {
         if (vp == param_overrides[i].object_ptr) {
             return param_overrides[i].value;
         }
     }
-    return *def_value_ptr;
+    if ((info.flags & AP_PARAM_FLAG_DEFAULT_POINTER) != 0) {
+        return *((float*)((ptrdiff_t)vp - info.def_value_offset));
+    }
+    return info.def_value;
+}
+
+float AP_Param::get_default_value(const AP_Param *vp, const struct Info &info)
```

Comment:
This is slightly upsetting in that is is exactly identical to the above method, just for `Info` rather than `GroupInfo`, could break out the flags, default and default offset into arguments....